### ISSUE TITLE
Error response from daemon: Cannot kill container: <id>: Container <id> is not running 

### ISSUE THREAD

**Issue by anuragpaliwal80** [chaos-mesh/chaos-mesh#1664]
## Bug Report

**What version of Kubernetes are you using?**

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:23:52Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"clean", BuildDate:"2021-01-13T13:15:20Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
**What version of Chaos Mesh are you using?**

`1.1.2` as well as in `1.0.1`

**What did you do?**

**What did you expect to see?**
controller logs:
```
2021-04-10T05:19:04.018Z	INFO	controllers.podchaos	Try to kill container	{"reconciler": "podchaos", "resource name": "ap-chaos/container-kill-example", "namespace": "ap-chaos", "podName": "dep1-5648b4dcbf-kq8k2", "containerID": "docker://1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c"}
2021-04-10T05:19:04.018Z	INFO	util	Creating client to chaos-daemon	{"node": "vmss-agent-worker0-11thrashva7-oqdgc000001"}
2021-04-10T05:19:04.022Z	ERROR	controllers.podchaos	kill container error	{"reconciler": "podchaos", "resource name": "ap-chaos/container-kill-example", "namespace": "ap-chaos", "podName": "dep1-5648b4dcbf-kq8k2", "containerID": "docker://1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c", "error": "rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c: Container 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c is not running"}
github.com/go-logr/zapr.(*zapLogger).Error
	/tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*endpoint).KillContainer
	/mnt/controllers/podchaos/containerkill/types.go:163
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*endpoint).Apply.func1
	/mnt/controllers/podchaos/containerkill/types.go:99
golang.org/x/sync/errgroup.(*Group).Go.func1
	/tmp/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57
2021-04-10T05:19:04.022Z	ERROR	controllers.podchaos	failed to kill container: kd1, pod: dep1-5648b4dcbf-kq8k2, namespace: ap-chaos	{"reconciler": "podchaos", "resource name": "ap-chaos/container-kill-example", "error": "rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c: Container 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c is not running"}
github.com/go-logr/zapr.(*zapLogger).Error
	/tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*endpoint).Apply.func1
	/mnt/controllers/podchaos/containerkill/types.go:101
golang.org/x/sync/errgroup.(*Group).Go.func1
	/tmp/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57
2021-04-10T05:19:04.022Z	ERROR	controllers.podchaos.ap-chaos/container-kill-example	fail to apply	{"reconciler": "podchaos", "resource name": "ap-chaos/container-kill-example", "error": "rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c: Container 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c is not running"}
github.com/go-logr/zapr.(*zapLogger).Error
	/tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/twophase.apply
	/mnt/controllers/twophase/state_machine.go:72
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*chaosStateMachine).run
	/mnt/controllers/twophase/state_machine.go:204
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*chaosStateMachine).Into
	/mnt/controllers/twophase/state_machine.go:208
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*Reconciler).Reconcile
	/mnt/controllers/twophase/types.go:115
github.com/chaos-mesh/chaos-mesh/pkg/router.(*Reconciler).Reconcile
	/mnt/pkg/router/reconciler.go:109
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:88
```

`k describe PodChaos -n ap-chaos` shows:
```
Failed Message:  rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c: Container 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c is not running
  Scheduler:
    Next Recover:  2021-04-10T05:19:04Z
    Next Start:    2021-04-10T05:19:04Z
Events:
  Type     Reason             Age                 From                 Message
  ----     ------             ----                ----                 -------
  Warning  ChaosInjectFailed  39m                 podchaos-controller  rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: a23d2d2e1e79f4e397f893e408fbb22f882801b643d817a0f6832ba0ed56fc77: Container a23d2d2e1e79f4e397f893e408fbb22f882801b643d817a0f6832ba0ed56fc77 is not running
  Warning  ChaosInjectFailed  37m                 podchaos-controller  rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 9cf76188ee2c066a0231abb970b0bb881ce806bdf57a00f30a59ec01f2e1a4ef: Container 9cf76188ee2c066a0231abb970b0bb881ce806bdf57a00f30a59ec01f2e1a4ef is not running
  Normal   ChaosRecovered     36m (x12 over 47m)  podchaos-controller  
  Warning  ChaosInjectFailed  35m (x2 over 35m)   podchaos-controller  rpc error: code = Unknown desc = Error response from daemon: Cannot kill container: 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c: Container 1f716ced2194fed497d8d8d0fd5e338a0251c8d62a6d1e8caf8115337239561c is not running
```

chaos experiment file:
```
apiVersion: chaos-mesh.org/v1alpha1
kind: PodChaos
metadata:
  name: container-kill
  namespace: ap-chaos
spec:
  action: container-kill
  mode: one
  containerName: 'kd1'
  selector:
    namespaces:
      - ap-chaos
    labelSelectors:
      'kd_binding': 'ap-chaos-binding'
  scheduler:
    cron: '@every 1m'

```

Once chaos resource is in `Failed` phase, even if the application pod start running smoothly (means its container is running), the chaos resorce does come back in action.

**What did you see instead?**

**Output of chaosctl**

**Comment by anuragpaliwal80** [comment]
One another sort of error with `conatiner-kill` action. Controller logs:

```
2021-04-10T06:04:17.732Z	INFO	controllers.PodChaos	Starting	{"reconciler": "podchaos", "action": "container-kill"}
2021-04-10T06:04:17.732Z	DEBUG	controller-runtime.manager.events	Warning	{"object": {"kind":"PodChaos","namespace":"ap-chaos","name":"container-kill","uid":"2856b847-e838-4d73-9d2b-3cea004dd21d","apiVersion":"chaos-mesh.org/v1alpha1","resourceVersion":"709818"}, "reason": "ChaosInjectFailed", "message": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.10.12.143:31767: i/o timeout\""}
2021-04-10T06:04:17.732Z	INFO	controllers.PodChaos	Chaos action:	{"reconciler": "podchaos", "action": "container-kill", "chaos": {"kind":"PodChaos","apiVersion":"chaos-mesh.org/v1alpha1","metadata":{"name":"container-kill","namespace":"ap-chaos","selfLink":"/apis/chaos-mesh.org/v1alpha1/namespaces/ap-chaos/podchaos/container-kill","uid":"2856b847-e838-4d73-9d2b-3cea004dd21d","resourceVersion":"709818","generation":3,"creationTimestamp":"2021-04-10T06:00:57Z","annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"chaos-mesh.org/v1alpha1\",\"kind\":\"PodChaos\",\"metadata\":{\"annotations\":{},\"name\":\"container-kill\",\"namespace\":\"ap-chaos\"},\"spec\":{\"action\":\"container-kill\",\"containerName\":\"kd1\",\"mode\":\"one\",\"scheduler\":{\"cron\":\"@every 1m\"},\"selector\":{\"labelSelectors\":{\"kd_binding\":\"ap-chaos-binding\"},\"namespaces\":[\"ap-chaos\"]}}}\n"},"managedFields":[{"manager":"kubectl-client-side-apply","operation":"Update","apiVersion":"chaos-mesh.org/v1alpha1","time":"2021-04-10T06:00:57Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:kubectl.kubernetes.io/last-applied-configuration":{}}},"f:spec":{".":{},"f:action":{},"f:containerName":{},"f:mode":{},"f:scheduler":{".":{},"f:cron":{}},"f:selector":{".":{},"f:labelSelectors":{".":{},"f:kd_binding":{}},"f:namespaces":{}}}}},{"manager":"chaos-controller-manager","operation":"Update","apiVersion":"chaos-mesh.org/v1alpha1","time":"2021-04-10T06:01:17Z","fieldsType":"FieldsV1","fieldsV1":{"f:status":{"f:experiment":{"f:phase":{}},"f:failedMessage":{}}}}]},"spec":{"selector":{"namespaces":["ap-chaos"],"labelSelectors":{"kd_binding":"ap-chaos-binding"}},"scheduler":{"cron":"@every 1m"},"action":"container-kill","mode":"one","value":"","containerName":"kd1","gracePeriod":0},"status":{"phase":"","failedMessage":"rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.10.12.143:31767: i/o timeout\"","scheduler":{},"experiment":{"phase":"Failed"}}}}
2021-04-10T06:04:17.732Z	INFO	controllers.PodChaos	Performing Action	{"reconciler": "podchaos", "action": "container-kill"}
2021-04-10T06:04:17.732Z	INFO	controllers.PodChaos	Try to kill container	{"reconciler": "podchaos", "action": "container-kill", "namespace": "ap-chaos", "podName": "dep1-5648b4dcbf-5q8qw", "containerID": "docker://9bd1463000853e04e079a7c3c4dfbf673aa11fc493bf8200f118fd974be1455a"}
2021-04-10T06:04:17.732Z	INFO	util	Creating client to chaos-daemon	{"node": "ip-10-10-12-143.ec2.internal"}
2021-04-10T06:04:37.732Z	ERROR	controllers.PodChaos	kill container error	{"reconciler": "podchaos", "action": "container-kill", "namespace": "ap-chaos", "podName": "dep1-5648b4dcbf-5q8qw", "containerID": "docker://9bd1463000853e04e079a7c3c4dfbf673aa11fc493bf8200f118fd974be1455a", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.10.12.143:31767: i/o timeout\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*Reconciler).KillContainer
	/src/controllers/podchaos/containerkill/types.go:163
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*Reconciler).Apply.func1
	/src/controllers/podchaos/containerkill/types.go:95
golang.org/x/sync/errgroup.(*Group).Go.func1
	/root/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57
2021-04-10T06:04:37.732Z	ERROR	controllers.PodChaos	failed to kill container: kd1, pod: dep1-5648b4dcbf-5q8qw, namespace: ap-chaos	{"reconciler": "podchaos", "action": "container-kill", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.10.12.143:31767: i/o timeout\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/podchaos/containerkill.(*Reconciler).Apply.func1
	/src/controllers/podchaos/containerkill/types.go:97
golang.org/x/sync/errgroup.(*Group).Go.func1
	/root/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57
2021-04-10T06:04:37.732Z	ERROR	controllers.PodChaos	failed to apply chaos action	{"reconciler": "podchaos", "action": "container-kill", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.10.12.143:31767: i/o timeout\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/twophase.applyAction
	/src/controllers/twophase/types.go:240
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*Reconciler).Reconcile
	/src/controllers/twophase/types.go:175
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).schedulePodChaos
	/src/controllers/podchaos/types.go:91
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).Reconcile
	/src/controllers/podchaos/types.go:52
github.com/chaos-mesh/chaos-mesh/controllers.(*PodChaosReconciler).Reconcile
	/src/controllers/podchaos_controller.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:88
2021-04-10T06:04:37.798Z	ERROR	controllers.PodChaos	unable to update chaos finalizers	{"reconciler": "podchaos", "action": "container-kill", "error": "Operation cannot be fulfilled on podchaos.chaos-mesh.org \"container-kill\": StorageError: invalid object, Code: 4, Key: /registry/chaos-mesh.org/podchaos/ap-chaos/container-kill, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 2856b847-e838-4d73-9d2b-3cea004dd21d, UID in object meta: "}
github.com/go-logr/zapr.(*zapLogger).Error
	/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/twophase.applyAction
	/src/controllers/twophase/types.go:248
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*Reconciler).Reconcile
	/src/controllers/twophase/types.go:175
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).schedulePodChaos
	/src/controllers/podchaos/types.go:91
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).Reconcile
	/src/controllers/podchaos/types.go:52
github.com/chaos-mesh/chaos-mesh/controllers.(*PodChaosReconciler).Reconcile
	/src/controllers/podchaos_controller.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:88
2021-04-10T06:04:37.801Z	ERROR	controllers.PodChaos	unable to update chaos status	{"reconciler": "podchaos", "action": "container-kill", "error": "Operation cannot be fulfilled on podchaos.chaos-mesh.org \"container-kill\": StorageError: invalid object, Code: 4, Key: /registry/chaos-mesh.org/podchaos/ap-chaos/container-kill, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: 2856b847-e838-4d73-9d2b-3cea004dd21d, UID in object meta: "}
github.com/go-logr/zapr.(*zapLogger).Error
	/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/twophase.updateFailedMessage
	/src/controllers/twophase/types.go:269
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*Reconciler).Reconcile
	/src/controllers/twophase/types.go:176
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).schedulePodChaos
	/src/controllers/podchaos/types.go:91
github.com/chaos-mesh/chaos-mesh/controllers/podchaos.(*Reconciler).Reconcile
	/src/controllers/podchaos/types.go:52
github.com/chaos-mesh/chaos-mesh/controllers.(*PodChaosReconciler).Reconcile
	/src/controllers/podchaos_controller.go:73
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/root/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/root/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:88
2021-04-10T06:04:37.801Z	INFO	controllers.PodChaos	pod chaos not found
```

This is from chaos version `1.0.1`

**Comment by anuragpaliwal80** [comment]
`StressChaos` is also having similar problem. Describe the chaos resource:

````
Status:
  Experiment:
    Phase:         Finished
  Failed Message:  rpc error: code = Unknown desc = container is not running, status: exited
  Scheduler:
Events:
  Type     Reason             Age                From                    Message
  ----     ------             ----               ----                    -------
  Warning  ChaosInjectFailed  11m (x2 over 11m)  stresschaos-controller  rpc error: code = Unknown desc = container is not running, status: exited
```

controller logs:
```
2021-04-11T04:29:05.881Z	ERROR	controllers.stresschaos.ap-chaos/burn-cpu-example	fail to step into the phase	{"reconciler": "stresschaos", "resource name": "ap-chaos/burn-cpu-example", "target phase": "Running", "error": "rpc error: code = Unknown desc = container is not running, status: exited"}
github.com/go-logr/zapr.(*zapLogger).Error
	/tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/twophase.(*Reconciler).Reconcile
	/mnt/controllers/twophase/types.go:117
github.com/chaos-mesh/chaos-mesh/pkg/router.(*Reconciler).Reconcile
	/mnt/pkg/router/reconciler.go:109
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.1-beta.0/pkg/util/wait/wait.go:88
```

Means once chaos resource is in Failed phase, even if the application container started running, the chaos resource does come back in action.

This is from chaos version `1.1.2`

**Comment by STRRL** [comment]
Errors about "container is not running" will occurs when the pod is at "ContainerCreating" phase. We are working on improving the logic. And you could wait until pods are "Running" before injecting the chaos.

**Comment by anuragpaliwal80** [comment]
Needs to check if this has been fixed/handled in 2.0 release.

**Comment by shanishmohan** [comment]
any suggestion on this error ?? i am trying to run network delay chaos

```
2021-09-01T07:26:42.277Z	INFO	trafficcontrol	traffic control Apply	{"namespace": "chaos", "name": "delay-webapp"}
2021-09-01T07:26:42.277Z	ERROR	records	fail to apply chaos	{"error": "unable to flush ip sets for pod webapp-6bcdd44597-tmch6"}
github.com/go-logr/zapr.(*zapLogger).Error
	/tmp/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/chaos-mesh/chaos-mesh/controllers/common.(*Reconciler).Reconcile
	/mnt/controllers/common/controller.go:170
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/tmp/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/tmp/go/pkg/mod/k8s.io/apimachinery@v0.17.0/pkg/util/wait/wait.go:88
2021-09-01T07:26:42.276Z	INFO	desiredphase	modify desiredPhase	{"desiredPhase": "Stop"}
2021-09-01T07:26:42.277Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "networkchaos-desiredphase", "request": "chaos/delay-webapp"}
```
